README.md

```diff
@@ -1,65 +1,82 @@
-   - Has a measurement API which exposes autoprofiling capabilities of the whole pipeline
-   - Pluggable measurements: The user can fit any logging mechanism to the profiler
+High performance:
+  - Multithreaded scheduler with worker pool
+  - In-place running in memory, allocations are only made during the setup phase, none are made during the running phase
+  - Uses mostly non-blocking algorithms avoiding OS locks
+
+Easy to use:
+  - Users don't need to have mutexes or any kind of synchronization primitives
+  - Nice, compact OOP API with templates to support custom data types
+  - Cross-platform C++11 code tested on gcc and clang
+
+Versatile:
+  - Configurable buffers to support low- and high delay units
+  - Supports stateless and stateful (accumulate) tasks
+  - Supports user controlled threads besides own thread pool to support UI, I/O or special thread affinity
+  - Supports circles (feedback) in the pipeline chain
+  - Supports multiple entry and exit (even reentry) points for distributed workflows
+
+Measurable (coming soon):
+  - Supports measurement-driven development via its own, minimal impact autoprofiling system
+  - Has a measurement API which exposes autoprofiling capabilities of the whole pipeline
+  - Pluggable measurements: The user can fit any logging mechanism to the profiler
+
+## Overview
+
+The scheduler manages a pool of worker threads to complete work in user-defined nodes on user-defined processing functions. Initially, when the scheduler is instantiated, it enters the setup phase. In this phase, channels should be created that represent the data processing stages in the pipeline using packet processing functions defined in the user's nodes. There can be two types of channels, scheduled or polled. The former channel type's packets are automatically scheduled to be processed by the worker pool, and the latter type's packets can be received manually be the user's own threads. This can be useful when the use case requires a dedicated thread for certain actions, such as when you need to process data using non-thread-safe librares, or when you need to make an exit point from the DSP pipeline to transmit some results out. When all channels are created, the scheduler can be started, at which point packets can start entering the pipeline.
+
+The processing functions defined in the user's nodes accept a packet to be processed as a parameter. After they are done processing a packet, packets can be sent onto the next channel (one that the user node has knowledge of) in the pipeline by acquiring a packet to be filled from that channel and filling it with the processed data. The filled packet is automatically queued for processing in the next channel upon the destruction of the stub beloning to the filled packet. The space in memory belonging to the processed packet is freed up for reuse automatically upon the destruction of the stub belonging to the packet.
```

Comment:
"belonging to the filled packet" instead of beloning

Reply:
That little typo managed to slip through, thanks.